src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -115,22 +116,15 @@ private void RegisterCore(IntPtr socketHandle, SocketAsyncContext context)
                 throw new InvalidOperationException(SR.net_sockets_handle_already_used);
             }
 
-            Interop.Error error = Interop.Sys.TryChangeSocketEventRegistration(_port, socketHandle, Interop.Sys.SocketEvents.None,
+            error = Interop.Sys.TryChangeSocketEventRegistration(_port, socketHandle, Interop.Sys.SocketEvents.None,
                 Interop.Sys.SocketEvents.Read | Interop.Sys.SocketEvents.Write, socketHandle);
             if (error == Interop.Error.SUCCESS)
             {
-                return;
+                return true;
             }
 
             _handleToContextMap.TryRemove(socketHandle, out _);
-            if (error == Interop.Error.ENOMEM || error == Interop.Error.ENOSPC)
-            {
-                throw new OutOfMemoryException();
-            }
-            else
-            {
-                throw new InternalException(error);
```

Comment:
I see we were previously throwing InternalException.  Ok, don't worry about it for your change.

Reply:
ok